### ISSUE TITLE
uMatrix does not block Facebook or Google on some sites

### ISSUE THREAD

**Issue by 48T-bird** [uBlockOrigin/uMatrix-issues#40]
### Prerequisites

- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uMatrix](https://old.reddit.com/r/uMatrix/)
- I tried to reproduce the issue when...
    - [x] uMatrix is the only extension--I removed all others when I added uMatrix. 
    - [x] uMatrix with default lists/settings
    - [x] using a new, unmodified browser profile
- [no, because Vista can't run the latest Firefox ] I am running the latest version of uMatrix that will work on this Firefox and on Opera
- [x] I checked the [documentation](https://github.com/gorhill/uMatrix/wiki) to understand that the issue I report is not a normal behavior

### Description

 On Meredith sites, Facebook and Google are not blocked. In my uMatrix settings, I have Google blocked everywhere, and Facebook blocked everywhere except on Facebook.com

### A specific URL where the issue occurs  
  win.fitnessmagazine.com
The same thing happens on all Meredith sweepstakes sites, and sometimes on their regular sites. (If you click on Rules, it will show you all the sweeps pages.)

### Steps to Reproduce

1. Go to the above page, or other pages from Meredith
2. [Second Step]  Sometimes uMatrix lists Facebook and Google, sometimes not, but they both always connect, according to TCP View (see screen shot).  I mean that sometimes uMatrix does not even see Facebook and Google on these sites, so doesn't list them.
3. [and so on...]

### Your environment

* uMatrix version: 1.0.0
* Browser Name and version: Firefox 52.6.0
    Opera 36.0.2130.80 with uMatrix version 1.3.0
My Internet connection is dialup, so I love uMatrix.  I'm a very experienced user with programs/add-ons like this.
--
* Operating System and version: Windows Vista Home SP2 (32 bit)

These are all related sites, through Meredith, a magazine publisher.  I haven't seen this happen on other sites. The screen shot that shows Facebook and Google allowed is from TCPView, a program from Sysinternals. In the logger, you can see that Google Analytics was blocked this time.  Facebook doesn't show up in the logger this time.  Sometimes neither show up in the logger or in the uMatrix list, but they always show connections in TCPView. No other pages are open. 

![umatrix on marthastewart com](https://user-images.githubusercontent.com/42113783/43686629-099ea66a-9897-11e8-9352-94bb0b275080.jpg)
![umatrix on fitnessmagazine](https://user-images.githubusercontent.com/42113783/43686996-65d0d8b8-989c-11e8-86a5-e41dce328eb0.jpg)
![umatrix logger on fitnessmagazine](https://user-images.githubusercontent.com/42113783/43686632-2d303328-9897-11e8-978f-c28b6ddfb8ea.jpg)
![umatrix allowing f and g](https://user-images.githubusercontent.com/42113783/43686642-55366b08-9897-11e8-8554-c9c3d2eb7424.jpg)

**Comment by uBlock-user** [comment]
That website goes onto a **infinite refresh loop**, so I can't investigate.

**Comment by gorhill** [comment]
I just `view-source:https://win.fitnessmagazine.com/` and this is what I find:

    <link rel="preconnect" href="https://staticxx.facebook.com" crossorigin>

It's a known [Chromium-specific issue](https://github.com/gorhill/uBlock/wiki/Dashboard:-Settings#important-note):

> On Chromium 51 and above (including browsers based on Chromium 51 and above), this setting does not block DNS lookups and preconnections.

When I wrote down that note in the wiki, I verified that this was not occurring with Firefox -- so you may want to switch to Firefox instead of a Chromium-based browser.

**Comment by gorhill** [comment]
Also, I see you are using uMatrix 1.0.0 -- it's no longer supported, I support only the webext version of uMatrix.

**Comment by uBlock-user** [comment]
Does Firefox also suffer from that issue ?

**Comment by gwarser** [comment]
On Firefox 52.9 AMO selected uMatrix 1.0.0 for installation. I see refresh loop.

**Comment by gorhill** [comment]
> OP's on Firefox

His TCP dump shows Opera as the source of the traffic, so he made the case with Opera, issue is the browser.

OP did not make the case Firefox + uMatrix 1.13.12 suffers same issue.

**Comment by gorhill** [comment]
> so doesn't that count ?

He is using uMatrix 1.0.0, it's unmaintained, and even though it was, the case would still not be made because there is no word about the values of `about:config` settings used by uMatrix 1.0.0 to prevent pre-fetch etc.

**Comment by gwarser** [comment]
I don't think uMatrix 1.0 even have pref for pre-fetch.

**Comment by gorhill** [comment]
> I don't think uMatrix 1.0 even have pref for pre-fetch.

It [does](https://github.com/gorhill/uMatrix/blob/master/platform/firefox/vapi-background.js#L240), and they are set by uMatrix, but these can be overridden at any time by the user or other extensions, and possibly the prefs.js (or whatever the name is).

**Comment by gwarser** [comment]
I meant checkbox in UI to switch on/off. I suppose this should be enabled automatically on install, it's what I see in latest version, but this does not work in 1.0.0 in Firefox ESR.

**Comment by 48T-bird** [comment]
Hello,
If you need anything else, I'll send it.  I just don't want people
thinking that they are blocking Facebook and Google when they may not be
blocked.  Those are probably the 2 sites that people most want blocked.
Thank you for your time.  uMatrix is definitely the best add-on I've ever
used.
Pat

On Sun, 05 Aug 2018 07:54:14 -0700 Saitama <notifications@github.com>
writes:
That website goes onto a infinite refresh loop, so I can't investigate.
�
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.
That is just today. It usually loads fine. Today, it's looping in both
Firefox and Opera.  Try tomorrow.  Actually, I just checked. They fixed
it and it's loading now.
Does Firefox also suffer from that issue ?  This is about Firefox. I did
list Opera because I use their main pages in Opera, but these sweeps
pages don't work in Opera.  I use Opera most because Firefox crashes so
often.  I probably shouldn't have listed Opera at all.

OP's on Firefox
His TCP dump shows Opera as the source of the traffic, so he made the
case with Opera, issue is the browser.
No, Opera is not the source, and not the cause of the trouble. The only
Opera listing is for Opera turbo, which helps for dialup.  I had Opera
open but was not using it.

All the Meredith sites come through resolute.com, which is why it's first
in the list.  Resolute doesn't show up in uMatrix though.  I've never
seen it there. Maybe that's a way Facebook and Google get through??  I
only know that they do connect, no idea why.
OP did not make the case Firefox + uMatrix 1.13.12 suffers same issue.
No, because this is Vista, which can't use a newer version of Firefox. 
uMatrix updates are set to "default," so, I assume this is the best
version of uMatrix that I can use. I just checked for updates for add-ons
and none were found. 
He is using uMatrix 1.0.0, it's unmaintained, and even though it was, the
case would still not be made because there is no word about the values of
about:config settings used by uMatrix 1.0.0 to prevent pre-fetch etc.
Also, I see you are using uMatrix 1.0.0 -- it's no longer supported, I
support only the webext version of uMatrix.

I was hoping you would test it with the latest Firefox and uMatrix to see
whether it still doesn't see Facebook and Google.  Other people may have
the same problem on other sites and not even know it.  Some day I'll be
able to get a new computer.  I recently reinstalled the OS and programs,
so I doubt I've changed many of the about:config settings in Firefox
except not to update automatically.  Checking now . . .
Every js preference in the about:config list is set at "default."
Every plugin preference is set at "default."
Here are the settings I found for uMatrix (in Firefox about:config):
1. extensions.umatrix.cloudStorage.myRulesPane;
2.
extensions.umatrix.placeholderBackground;url("data:image/png;base64,iVBOR
w0KGgoAAAANSUhEUgAAAAoAAAAKCAAAAACoWZBhAAAABGdBTUEAALGPC/xhBQAAAAJiS0dEAP
+Hj8y/AAAAB3RJTUUH3wwIAAgyL/YaPAAAACJJREFUCFtjfMbOAAQ/gZiFnQPEBAEmGIMIJgt
IL8QEgtoAIn4D/96X1KAAAAAldEVYdGRhdGU6Y3JlYXRlADIwMTUtMTItMDhUMDA6MDg6NTAr
MDM6MDAasuuJAAAAJXRFWHRkYXRlOm1vZGlmeQAyMDE1LTEyLTA4VDAwOjA4OjUwKzAzOjAwa
+9TNQAAAABJRU5ErkJggg==") repeat scroll #fff
3. extensions.umatrix.placeholderBorder;1px solid rgba(0, 0, 0, 0.05)
4. extensions.umatrix.placeholderDocument;<html><head><meta
charset="utf-8"><style>body { background: {{bg}};color: gray;font: 12px
sans-serif;margin: 0;overflow: hidden;padding: 2px;white-space: nowrap;}a
{ color: inherit;padding: 0 3px;text-decoration:
none;}</style></head><body><a href="{{url}}" title="{{url}}"
target="_blank">&#x2191;</a>{{url}}</body></html>
5.
extensions.umatrix.placeholderImage;data:image/gif;base64,R0lGODlhAQABAAA
AACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==
6. extensions.umatrix.popupHideBlacklisted;false
7.
services.sync.prefs.sync.extensions.umatrix.cloudStorage.myRulesPane;true


I don't think uMatrix 1.0 even have pref for pre-fetch.
It does, and they are set by uMatrix, but these can be overridden at any
time by the user or other extensions, and possibly the prefs.js (or
whatever the name is).
This only happens on the Meredith sites as far as I've found.  uMatrix
works great on all other sites.  The other extensions I have are "https
everywhere," one for IP's in the address bar, Status-4-Evar, NoSquint.
Wouldn't those settings make this happen on every site?  I didn't see any
prefs.js

I meant checkbox in UI to switch on/off. I suppose this should be enabled
automatically on install, it's what I see in latest version, but this
does not work in 1.0.0 in Firefox ESR.

I haven't seen such a switch, and only use uMatrix to block/allow certain
sites, regular use.

**Comment by gwarser** [comment]
In Firefox, check if these values are set:

```
network.prefetch-next;false
network.http.speculative-parallel-limit;0
```

if not then set them and redo your tests.

**Comment by 48T-bird** [comment]
These are the default settings in Firefox about:config

network.prefetch-next;true
network.http.speculative-parallel-limit;6

I reset them to the "user set" values that you sent.  I cleared Firefox
cookies, history, cache, etc. between each trial.  I opened the page from
bookmarks.

This is what showed in TCPView. 
X1Exec.exe is my ISP. 
Node.exe is Prey, a program that can help find a stolen computer. 
The TCPV6 pops up on the Meredith sites and downloads a lot, but I don't
know what it is. 
Resolute is their main server
Cloudfront.net is also theirs
198.186.175.53 downloads the pictures for their sites
This time, Facebook didn't connect


If I turn either one of the preferences back to default, Facebook
connects.  No matter how the preferences are set, Facebook does not show
in the uMatrix list. This screen shot is with the preferences set to
default, but it is the same with the preferences set as you requested.
Resolute is never in the list either.


Facebook still connects when the preferences are set to default:


1e100.net always connects, no matter how the preferences are set:
   Name Server: NS1.GOOGLE.COM
   Name Server: NS2.GOOGLE.COM
   Name Server: NS3.GOOGLE.COM
   Name Server: NS4.GOOGLE.COM

I hope you understand all this.  Does everyone who uses uMatrix have to
change those preferences?  Just wondering; I don't need an answer.

Now I tried one more thing--setting Facebook as blocked completely.  When
I go to facebook.com, it says "uMatrix has prevented the following page
from loading: https://www.facebook.com/"  But when I open a Meredith
page, Facebook still connects (with preferences set to default).  Yes, my
brain is fried for tonight.

Thank you for all your time, and for uMatrix.
Pat


On Wed, 08 Aug 2018 14:50:43 -0700 gwarser <notifications@github.com>
writes:
In Firefox, check if these values are set:
network.prefetch-next;false
network.http.speculative-parallel-limit;0

if not then set them and redo your tests.
�

**Comment by gwarser** [comment]
@48T-bird please respond directly on Github, your messages are _unreadable_, images are missing.

>If I turn either one of the preferences back to default, Facebook connects. No matter how the preferences are set, Facebook does not show in the uMatrix list.

This means these TCP connections are only for prefetching. Caused by [`link prefetch`](https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types) tags:

>Provides a hint to the browser suggesting that it open a connection to the linked web site in advance, without disclosing any private information or downloading any content, so that when the link is followed the linked content can be fetched more quickly.

Nothing is transferred (no http request) - nothing in logs.

Looks like there was a bug in uMatrix 1.0.0 because these about:config preferences should be set on install. Legacy uMatrix is no longer maintained - you need to set these settings by yourself.

**Comment by 48T-bird** [comment]
(Posting the last message directly so the pictures will show.)
Understood. The new version sets those preferences on install, so that fixes this problem (which looked like a connection, but actually no tracking was happening).  
I'll be glad when I can get a new computer . . . but for now, fixing my truck has to come first.
Thank You.
**********************************************************
These are the default settings in Firefox about:config
 
network.prefetch-next;true
network.http.speculative-parallel-limit;6
 
I reset them to the "user set" values that you sent.  I cleared Firefox cookies, history, cache, etc. between each trial.  I opened the page from bookmarks.
 
This is what showed in TCPView. 
X1Exec.exe is my ISP. 
Node.exe is Prey, a program that can help find a stolen computer. 
The TCPV6 pops up on the Meredith sites and downloads a lot, but I don't know what it is. 
Resolute.com is their main server
Cloudfront.net is also theirs
198.186.175.53 downloads the pictures for their sites
This time, Facebook didn't connect
![umatrix 2 tcpview](https://user-images.githubusercontent.com/42113783/43898346-890df92e-9bac-11e8-9f70-6edaa9c595d7.jpg)

 
If I turn either one of the preferences back to default, Facebook connects.  No matter how the preferences are set, Facebook does not show in the uMatrix list. This screen shot is with the preferences set to default, but it is the same with the preferences set as you requested. Resolute is never in the list either.
![umatrix 2](https://user-images.githubusercontent.com/42113783/43898483-fd4e6ee0-9bac-11e8-9bb0-3e84799eaf29.jpg)

 Facebook still connects when the preferences are set to default:
![umatrix 2b tcpview ff set to default](https://user-images.githubusercontent.com/42113783/43898591-4ba3455c-9bad-11e8-97a4-053e62ca52bf.jpg)

1e100.net always connects, no matter how the preferences are set:
   Name Server: NS1.GOOGLE.COM
   Name Server: NS2.GOOGLE.COM
   Name Server: NS3.GOOGLE.COM
   Name Server: NS4.GOOGLE.COM

I hope you understand all this.  Does everyone who uses uMatrix have to change those preferences?  Just wondering; I don't need an answer.
 
Now I tried one more thing--setting Facebook as blocked completely.  When I go to facebook.com, it says "uMatrix has prevented the following page from loading: https://www.facebook.com/"  But when I open a Meredith page, Facebook still connects (with preferences set to default).  Yes, my brain is fried for tonight.
 
Thank you for all your time, and for uMatrix.
Pat
